tflearn/helpers/trainer.py

```diff
@@ -87,8 +87,12 @@ def __init__(self, train_ops, graph=None, clip_gradients=5.0,
             self.validate_trainop_names()
 
             self.global_loss = None
-            self.global_step = tf.Variable(0., name='Global_Step',
-                                           trainable=False)
+            if global_step:
+                self.global_step = global_step
+            else:
```

Comment:
Sure, that makes sense. I was thinking more like we don't need the entire if else. It can be something like:
```
if global_step is None:
    global_step = tf.Variable(0., name='Global_Step', trainable=False)
self.global_step = global_step
```

but your way is more explicit.